onnxruntime/core/providers/cuda/tensor/tile_impl.cu

```diff
@@ -64,9 +74,12 @@ void TileMemcpyImpl(
     const size_t num_input_elements,
     T* output_data,
     const size_t num_output_elements) {
-  int blocksPerGrid = (int)(ceil(static_cast<float>(num_output_elements) / GridDim::maxThreadsPerBlock));
+  // each block has 256 threads, total block will be ceil(num_input / 256) >> namely, each thread takes care one input element
+  // each cuda thread processes num_output/num_input output data elements
+  int blocksPerGrid = (int)(ceil(static_cast<float>(num_input_elements) / GridDim::maxThreadsPerBlock));
```

Comment:
Curious how this is making it faster ? In this approach, one CUDA thread does more work and we will limit the number of blocks launched because number of blocks is now computed based on number of input elements. This could possibly improve perf if the number of blocks computed previously was too high. Can you share more info such as the number of blocks before and after this change in your model ? 

Your logic also adds an extra `if` within the loop to ensure we are within the output array bounds. I am not sure if this is guaranteed to be faster for all use-cases. This optimization was added for a 1PP model and we should be vary of any regressions this might cause.

Reply:
thx for reviewing.
the reason why it could be faster is because gpu's global mem latency is quite high, in this way we only copy data once and reuse it. And normally, the number of input will be high enough to have enough block to occupy all gpu SMs(128 around).

Reply:
@SherlockNoMad  do we have any ci pipeline to avoid perf regression?

Reply:
@hariharans29 the perf gain in my env.
![image](https://user-images.githubusercontent.com/43435212/127987122-d7fc5f95-fda3-4584-8306-d75bffc2e31b.png)

Reply:
in my model, the input shape is 3072*256. so it could have larger gain.

Reply:
Thanks for the explanation and the perf numbers.